android/app/src/main/java/org/uwaterloo/subletr/pages/chat/ChatListingPageView.kt

```diff
@@ -0,0 +1,212 @@
+			verticalArrangement = Arrangement.Center,
+		) {
+			CircularProgressIndicator()
+		}
+	} else if (uiState is ChatListingPageUiState.Loaded) {
+		Scaffold(
+			modifier = modifier,
+			topBar = {
+				Row(
+					modifier = Modifier
+						.fillMaxWidth(1.0f)
+						.padding(
+							start = dimensionResource(id = R.dimen.s),
+						),
+					verticalAlignment = Alignment.CenterVertically,
+				) {
+					Text(
+						text = stringResource(id = R.string.messages),
+						style = MaterialTheme.typography.titleMedium,
+					)
+				}
+			},
+			content = { padding: PaddingValues ->
+				if (uiState.contacts.isEmpty()) {
+					Column(
+						modifier = modifier
+							.fillMaxSize()
+							.padding(padding)
+							.background(Color.Transparent),
+						horizontalAlignment = Alignment.CenterHorizontally,
+						verticalArrangement = Arrangement.Center,
+					) {
+						Text(
+							stringResource(id = R.string.empty_chat),
+							color = primaryTextColor,
+						)
+					}
+				} else {
+					LazyColumn(modifier = modifier.padding(padding)) {
+						items(items = uiState.contacts) { contact: Contact ->
+							ListItem(contact = contact)
+						}
+					}
+				}
+			}
+		)
+	}
+}
+
+@Composable
+fun ListItem(contact : Contact) {
+	val name = contact.name
+	val lastMsg = contact.msg.last()
+	Box(
+		modifier = Modifier
+			.padding(
+				vertical = dimensionResource(id = R.dimen.xs),
+				horizontal = dimensionResource(id = R.dimen.s),
+			)
+			.wrapContentHeight()
+			.fillMaxWidth(1.0f)
+			.clip(
+				RoundedCornerShape(
+					dimensionResource(id = R.dimen.s)
+				)
+			)
+			.background(secondaryButtonBackgroundColor),
+	) {
+		Row(verticalAlignment = Alignment.CenterVertically) {
+			Icon(name = name)
+			Column(
+				modifier = Modifier
+					.weight(1f)
+					.padding(vertical = dimensionResource(id = R.dimen.s)),
+				verticalArrangement = Arrangement.Center,
+			) {
+				Text(text = name,
+					style = MaterialTheme.typography.titleSmall)
+				Text(text = lastMsg,
+					style = MaterialTheme.typography.bodySmall)
+			}
+			if (contact.unread > 0) {
+				Notification(contact.unread)
+			}
+			
+		}
+	}
+}
+
+@Composable
+fun Icon(name : String) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.s))
+			.clip(CircleShape),
+	) {
+		Box(
+			modifier = Modifier
+				.width(dimensionResource(id = R.dimen.xxl))
+				.height(dimensionResource(id = R.dimen.xxl))
+				.background(subletrPink),
+			contentAlignment = Alignment.Center,
+		) {
+			Text(
+				text = name.first().toString(),
+				style = MaterialTheme.typography.titleSmall,
+				color = textOnSubletrPink,
+			)
+		}
+	}
+}
+
+@Composable
+fun Notification(n : Int = 1) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.s))
+			.clip(CircleShape),
+	) {
+		Box(
+			modifier = Modifier
+				.width(dimensionResource(id = R.dimen.l))
+				.height(dimensionResource(id = R.dimen.l))
+				.background(notificationGreen),
```

Comment:
Looks like this should be pink?

Reply:
notification circle should be green?

Reply:
The figma notification circle is pink, not green.
![image](https://github.com/andr-dev/CS446/assets/38482987/7c363522-d15e-4725-b488-a5039b1763eb)
Again, you should be referencing the figma for UI design details. Abhay spent a long time making these so that the app looks consistent and clean across pages. We should not be making arbitrary decisions.